File: CSharp_Unity/Chess/Assets/Code/Domain/MoveResult.cs
using System.Collections.Generic;

namespace ChessKata.Domain
{
    public class MoveResult
    {
        public MoveResult(IEnumerable<RuleViolation> violations, GameState gameState)
        {
            Violations = violations;
            GameState = gameState;
        }

        public IEnumerable<RuleViolation> Violations { get; }
        public GameState GameState { get; }
    }
}

File: Kotlin/src/main/kotlin/org/chess/ruleengine/MoveResult.kt
package org.chess.ruleengine

import org.chess.domain.GameState
import org.chess.domain.RuleViolation

data class MoveResult(val gameState: GameState, val violations: Collection<RuleViolation>) {

    // Temporary backwards compatibility for usages from Java code that expect
    // an implicit get method from a Java record.
    fun gameState() = gameState
    fun violations() = violations
}
